rules/java-net/src/main/resources/otagent/java-net.btm

```diff
@@ -0,0 +1,110 @@
+
+RULE java-net: Java HttpUrlConnection connect entry
+CLASS ^java.net.HttpURLConnection
+METHOD connect()
+HELPER io.opentracing.contrib.agent.OpenTracingHelper
+AT ENTRY
+IF !$0.connected && retrieveSpan($0) == null && !ignore($0)
+DO
+  associateSpan($0, getTracer().buildSpan($0.getRequestMethod())
+    .withTag(io.opentracing.tag.Tags.SPAN_KIND.getKey(), io.opentracing.tag.Tags.SPAN_KIND_CLIENT)
+    .withTag(io.opentracing.tag.Tags.HTTP_URL.getKey(), $0.getURL().toString())
+    .withTag("http.query", $0.getURL().getQuery())
+    .asChildOf(currentSpan())
+    .start());
+  getTracer().inject(retrieveSpan($0).context(), io.opentracing.propagation.Format$Builtin.HTTP_HEADERS,
+    new io.opentracing.contrib.agent.jnet.propagation.HttpURLConnectionInjectAdapter($0));
+ENDRULE
```

Comment:
My initial aim has been to keep the helper class as simple as possible, primarily providing access to the Tracer, and then adding mechanisms as they were required. So for this initial release I would prefer to keep it that way, but then over time see what general mechanisms should be pushed up to the helper.

So would it be ok if I create a separate issue to discuss this further?